### PR TITLE
NAPSSPO-596 - update tssc workflow to colapse deploy and create-deployment-environment steps

### PR DESCRIPTION
**Pull request by itewk**
* NAPSSPO-596 - update tssc workflow to colapse deploy and create-deployment-environment steps
* initial start of adding an TSSC opinions module.
* update language from "tool categories" to "components"

For rendered version see: https://itewk.github.io/tsc-docs/

### CODE REVIEW

**Comment by itewk** [comment]
@lcurry i snuck in the framework for adding in TSSC opinions to the doc. this should give you the place to add in more based on our conversation.

**Review by kfrankli — DISMISSED** [review]
ship it

File: docs/modules/con-tssc-opinions.adoc [hunk 1]
```diff
@@ -0,0 +1,16 @@
+[id="{ProjectNameID}-opinions", reftext="{ProjectShortName} Opinions"]
+= {ProjectShortName} Opinions
+
+The TSSC is an opinionated software development and deployment workflow. The key being, opinionated. Which means it is filled with opinions, some direct, some less so. This document is meant to capture all of those opinions which feed into the design and implementation of the TSSC.
```
Comment (HunterGerlach) [inline docs/modules/con-tssc-opinions.adoc]: Should TSSC be directly referenced here or should it be a variable ref? @itewk

File: docs/modules/ref-tssc-workflow-tools.adoc [hunk 2]
```diff
@@ -1,17 +1,17 @@
...
+Once a <<{ProjectNameID}-workflow-{context}>> is decided upon, then the next task is to determine all of the types of components needed to implement the workflow, and then finally, the actual component implementers (aka, specific technology implementations and tools). This section contains both the component types and their purposes as well as opinions on which actual component implementations to use to fullfil the required components needs.
 
 [id="{ProjectNameID}-workflow-tool-purposes-{context}"]
 == Tool Categories and Purposes
 
-This section describes the category of tools need to implement the {ProjectShortName} and their purpose in doing so.
+This section describes the components need to implement the {ProjectShortName} and their purpose in doing so.
```
Comment (HunterGerlach) [inline docs/modules/ref-tssc-workflow-tools.adoc]: @itewk Change "need" to "needed"

File: docs/modules/con-tssc-opinions.adoc (hunk 1 above)
Comment (itewk) [reply docs/modules/con-tssc-opinions.adoc]: it should be a variable.....

File: docs/modules/ref-tssc-workflow-tools.adoc (hunk 2 above)
Comment (itewk) [reply docs/modules/ref-tssc-workflow-tools.adoc]: thanks

**Review by HunterGerlach — CHANGES_REQUESTED** [review]
Some updates needed.

I couldn't add a comment to the images, but "continues" needs to be updated to "continuous" (see attached)
Screenshot from 2020-08-07 14-22-52

File: docs/modules/ref-tssc-workflow-tools.adoc (hunk 2 above)
Comment (itewk) [reply docs/modules/ref-tssc-workflow-tools.adoc]: fixed

File: docs/modules/con-tssc-opinions.adoc (hunk 1 above)
Comment (itewk) [reply docs/modules/con-tssc-opinions.adoc]: fixed.

**Comment by itewk** [comment]
@HunterGerlach fixed the image now too.

File: docs/modules/con-tssc-opinions.adoc [hunk 3]
```diff
@@ -0,0 +1,16 @@
+[id="{ProjectNameID}-opinions", reftext="{ProjectShortName} Opinions"]
+= {ProjectShortName} Opinions
+
+The {ProjectName} (ProjectShortName) is an opinionated software development and deployment workflow. The key being, opinionated. Which means it is filled with opinions, some direct, some less so. This document is meant to capture all of those opinions which feed into the design and implementation of the TSSC.
```
Comment (HunterGerlach) [inline docs/modules/con-tssc-opinions.adoc]: Still need to update the last TSSC var. @itewk

**Review by HunterGerlach — CHANGES_REQUESTED** [review]
The image is still not quite right. Do a global search for "continues" and replace it with "continuous".

File: docs/modules/con-tssc-opinions.adoc (hunk 3 above)
Comment (itewk) [reply docs/modules/con-tssc-opinions.adoc]: fixed.

**Review by HunterGerlach — DISMISSED** [review]
Looks good!

**Review by lcurry — CHANGES_REQUESTED** [review]
In "Opinions" section should  "{ProjectShortName}" instead of "(ProjectShortName)" ?

**Comment by itewk** [comment]
@lcurry thanks for some of the opinon updates, i meant to just be creating hte framework for you to be able to add in more later in your own MR, but we can sneak some of that in now. I did do some refacotring on it to not conflate the use of fork & branch and the type of source control development flow. aka, fork & branch v branch does not preclude or require git-flow. you can do git-flow using either type of branching stratigy

File: docs/modules/ref-tisc-workflow-components.adoc [hunk 4]
```diff
@@ -11,8 +11,8 @@ The following table lists chosen tooling for the TISC MVP implementation.  A sep
 .Tooling by Workflow Implementation
 [cols="a,a,a",options="header"]
 |===
-| Purpose
-2+| Tools by Workflow Type
+| Component Type
+2+| Implementing Tools by Workflow Type
 
 |
 h| *Role/Module*
```
Comment (lcurry) [inline docs/modules/ref-tisc-workflow-components.adoc:18]: do we need the two-tiered headers on Table 5? "Role/Module" and "Stack" complicate table and seem to provide limited value

**Review by lcurry — APPROVED** [review]

**Review by kfrankli — APPROVED** [review]

**Pull request merged**